app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -1,51 +1,179 @@
         // If no userId is provided, we get the user that is currently logged in.
-        val userId = when(args.userId) {
-                null, "" -> currentUserProvider.getCurrentUserId()
-                else -> args.userId
+        shownUserProfileId = when(args.userId) {
+            null, "" -> currentUserId
+            else -> args.userId
         }
+        userViewModel.setUser(shownUserProfileId)
+        imageAccess = ImageAccess(currentActivity)
+        imageAccess.setupImageView(profilePicture)
+        lifecycle.addObserver(imageAccess)
+        userViewModel.user.observe(viewLifecycleOwner, { user ->
+            displayUserFields(user)
+        })
+        binding.viewModel = userViewModel
+        loggedInUserEmail = currentUserProvider.getCurrentUserEmail()
+        initSetup()
+        setupViewAndButtonsAction()
+        return binding.root
+    }
 
-        userViewModel.setUser(userId)
+    private fun initSetup() {
+        val fields = listOf(
+            binding.imageView,
+            binding.gallerycameraholder,
+            binding.nameText,
+            binding.textEmail,
+            binding.applyholder
+        )
+        for(view: View in fields) {
+            view.visibility = View.GONE
+        }
+        binding.textEmail.text = getString(R.string.text_email_unavailable)
+    }
 
+    private fun setupButtonsVisibility() {
+        currentUserId = currentUserProvider.getCurrentUserId()
+        val pictureButtonsRow = binding.gallerycameraholder
+        if(currentUserId != null && isAuthUserDisplayedUser()) {
+            pictureButtonsRow.visibility = View.VISIBLE
+        }
+    }
 
-        userViewModel.user.observe(viewLifecycleOwner, { user ->
-            if (user != null) {
-                binding.nameText.text = user.name
-                Glide.with(this).load(user.profilePicture).into(binding.imageView)
+    private fun setupButtonsAction() {
+        val buttons = listOf(binding.btnApply, binding.btnOpenCamera, binding.btnOpenGallery)
+        for(button: Button in buttons) {
+            button.setOnClickListener {
+                getAction(button)
             }
-        })
+        }
+    }
 
-        binding.viewModel = userViewModel
+    private fun setEmailText() {
+        val emailText = binding.textEmail
+        if(isAuthUserDisplayedUser() && currentUserId != null) {
+            emailText.text = loggedInUserEmail
+        }
+        if(currentUserId != null) {
+            emailText.visibility = View.VISIBLE
+        }
+    }
 
-        return binding.root
+    private fun setupTopInfoVisibility() {
+        val topInfoText = binding.upfTopinfo
+        if(currentUserId != null) {
+            topInfoText.visibility = View.GONE
+        }
+    }
+
+    private fun isAuthUserDisplayedUser(): Boolean {
+        val shownUserId = args.userId
+        return shownUserId == null || shownUserId == currentUserId
+    }
+
+    private fun setupPfpAndNameVisibility() {
```

Comment:
What is "Pfp"? Might rename it to make it clearer

Reply:
I concur